jina/types/request/__init__.py

```diff
@@ -230,10 +234,11 @@ def as_response(self):
         """
         Return a weak reference of this object but as :class:`Response` object. It gives a more
         consistent semantics on the client.
+
+        :return: `self` as an instance of `Response`
         """
-        base_cls = self.__class__
-        base_cls_name = self.__class__.__name__
-        self.__class__ = type(base_cls_name, (base_cls, Response), {})
+        print(f' as_respons')
+        return Response(self)
```

Comment:
You are not passing `envelope`.

What brings me to the question: Why is `envelope` anyhow a field of a request? It should only be part of a message. Strange :thinking:

Reply:
I just realized, it is there to extract compression algorithm, so changed to pass `compression` algorithm instead